externs/firebase-app-externs.js

```diff
@@ -273,3 +273,38 @@ firebase.Promise.reject = function(error) {};
  * @return {!firebase.Promise<!Array<*>>}
  */
 firebase.Promise.all = function(values) {};
+
+/** @interface */
+firebase.Observer = function() {};
+
+/**
+ * @template V
+ * @param {?V} value
+ */
+firebase.Observer.prototype.next = function(value) {};
+
+/**
+ * @template E
+ * @param {!E} error
+ */
+firebase.Observer.prototype.error = function(error) {};
+
+firebase.Observer.prototype.complete = function() {};
+
+/**
```

Comment:
Ah well, too bad. I removed the templates from NextFn and ErrorFn and moved the typedefs back to the INTERNAL file, where they're still used. I replaced all the other uses with the original `!function(...)` annotations.